kafka/src/test/groovy/io/micronaut/configuration/kafka/exceptions/DefaultKafkaListenerExceptionHandlerSpec.groovy

```diff
@@ -0,0 +1,163 @@
+package io.micronaut.configuration.kafka.exceptions
+
+import io.micronaut.context.annotation.Property
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import org.apache.kafka.clients.producer.ProducerConfig
+import io.micronaut.configuration.kafka.AbstractEmbeddedServerSpec
+import io.micronaut.configuration.kafka.annotation.KafkaClient
+import io.micronaut.configuration.kafka.annotation.KafkaListener
+import io.micronaut.configuration.kafka.annotation.Topic
+import io.micronaut.context.annotation.Requires
+import org.apache.kafka.common.TopicPartition
+
+import static io.micronaut.configuration.kafka.annotation.OffsetReset.EARLIEST
+import static io.micronaut.configuration.kafka.annotation.OffsetStrategy.DISABLED
+
+class DefaultKafkaListenerExceptionHandlerSpec extends AbstractEmbeddedServerSpec {
+
+    private static final String TOPIC_SEEK = "on-deserialization-error-seek"
+    private static final String TOPIC_COMMIT = "on-deserialization-error-commit"
+    private static final String TOPIC_NOTHING = "on-deserialization-error-do-nothing"
+
+    void "test seek past record on deserialization error by default"() {
+        given:
+        StringProducer stringProducer = context.getBean(StringProducer)
+        DefaultBehaviorOnDeserializationErrorConsumer consumer = context.getBean(DefaultBehaviorOnDeserializationErrorConsumer)
+
+        when: "A producer sends a message with wrong serialization"
+        stringProducer.sendToSeekTopic("not-a-uuid")
+
+        then: "The message is skipped with a seek() but not committed"
+        conditions.eventually {
+            consumer.currentPosition == 1
+            consumer.committedOffset == 0
+        }
+    }
+
+    void "test commit record on deserialization error"() {
+        given:
+        StringProducer stringProducer = context.getBean(StringProducer)
+        CommitOnDeserializationErrorConsumer consumer = context.getBean(CommitOnDeserializationErrorConsumer)
+
+        when: "A producer sends a message with wrong serialization"
+        stringProducer.sendToCommitTopic("not-a-uuid")
+
+        then: "The message is skipped and committed"
+        conditions.eventually {
+            consumer.currentPosition == 1
+            consumer.committedOffset == 1
+        }
+    }
+
+    void "test do nothing on deserialization error"() {
+        given:
+        StringProducer stringProducer = context.getBean(StringProducer)
+        DoNothingOnDeserializationErrorConsumer consumer = context.getBean(DoNothingOnDeserializationErrorConsumer)
+
+        when: "A producer sends a message with wrong serialization"
+        stringProducer.sendToDoNothingTopic("not-a-uuid")
+
+        then: "The message is skipped, but not committed"
+        conditions.eventually {
+            consumer.currentPosition > 0
```

Comment:
@guillermocalvo it's strange that this test passes, the `DoNothingOnDeserializationErrorConsumer` explicitly sets the `skipRecordOnDeserializationFailure` to false so I'd expect the offset not to be advanced... or am I missing something?

Reply:
@giamo Previous behavior was fixed by: https://github.com/micronaut-projects/micronaut-kafka/pull/771